alacritty_terminal/src/renderer/rects.rs

```diff
@@ -36,16 +36,46 @@ impl RenderRect {
     }
 }
 
-struct RenderLine {
-    start: Point,
-    end: Point,
-    color: Rgb,
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct RenderLine {
+    pub start: Point,
+    pub end: Point,
+    pub color: Rgb,
 }
 
 impl RenderLine {
-    fn into_rect(self, flag: Flags, metrics: &Metrics, size: &SizeInfo) -> RenderRect {
-        let start_x = self.start.col.0 as f32 * size.cell_width;
-        let end_x = (self.end.col.0 + 1) as f32 * size.cell_width;
+    pub fn into_rects(
+        mut self,
+        flag: Flags,
+        metrics: &Metrics,
+        size: &SizeInfo,
+    ) -> Vec<RenderRect> {
+        let mut rects = Vec::new();
+
+        while self.start.line < self.end.line {
+            let mut end = self.start;
+            end.col = size.cols() - 1;
+            rects.push(Self::create_rect(metrics, size, flag, self.start, end, self.color));
+
+            self.start.col = Column(0);
+            self.start.line += 1;
+        }
+
+        rects.push(Self::create_rect(metrics, size, flag, self.start, self.end, self.color));
+
+        rects
+    }
+
+    fn create_rect(
+        metrics: &Metrics,
```

Comment:
Why does RenderLine has a static method, called `create_rect`, which return `RenderRect`? I think it should be moved to RenderRect, right?

Reply:
bump

Reply:
Oh, this actually exists for a pretty good reason.

Previously all this code was inside the `as_rects` method, but the code is now executed multiple times with different parameters. That's why it can't take `self` either, that would be misleading as it takes modified parameters that are stored on `self` in a different fashion.

So the only option would be moving it to `RenderRect`. However I have a hard time coming up with a name for that, which is usually an indication that it's not the perfect place for it either. The parameters are just all `RenderLine` exclusive and it acts purely as a way to deduplicate code.

Reply:
I do understand the reason behind it, since we've already walk though this discussion in a past when we were refactoring/fixing underlines, and I do remember that there were similar options. So we can leave it as it is now, and promise to never make it public in its current state.

Reply:
Oh yeah, if we would want to make this public it should be moved onto `RenderRect`.